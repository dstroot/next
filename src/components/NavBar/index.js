import React, { useState, useRef, useEffect } from 'react';
import { Link } from '@reach/router';

// animations
import { TimelineMax } from 'gsap';
import ScrollMagic from 'scrollmagic';
import 'scrollmagic/scrollmagic/uncompressed/plugins/animation.gsap';
import 'scrollmagic/scrollmagic/uncompressed/plugins/debug.addIndicators';

// components
import HamButton from '../HamButton';

// images
import logo from './media/NXT_Logo_light.svg';

// hooks
import useMediaQuery from '../../hooks/useMediaQuery';

// isCurrent - true if the location.pathname is exactly the same as the anchorâ€™s href.
export const isActive = ({ isCurrent }) => {
  return isCurrent
    ? { className: 'active nav-link' }
    : { className: 'nav-link' };
};

// getProps` Calls up to you to get props for the underlying anchor element.
// Useful for styling the anchor as active.
const ExactNavLink = props => <Link getProps={isActive} {...props} />;

const NavBar = () => {
  const small = useMediaQuery(`screen and (max-width: 575px)`);
  const [clicked, setClicked] = useState(false);

  const toggle = event => {
    setClicked(!clicked);
    window.scrollTo(0, 0);
  };

  const [animation, setAnimation] = useState(null);

  // GSAP
  let tl = new TimelineMax();

  // ScrollMagic
  let controller = useRef(null);
  controller = new ScrollMagic.Controller();

  useEffect(() => {
    setAnimation(
      // GSAP
      tl.to('.navbar', 0.5, {
        css: {
          borderBottom: '1px solid rgba(200, 200, 200, 1)',
          backgroundColor: '#ffffff',
        },
      })
    );
    setAnimation(
      // ScrollMagic
      new ScrollMagic.Scene({
        triggerElement: '#trig',
        triggerHook: 0.2,
        offset: 100,
      })
        .setTween(tl)
        // .addIndicators() // for debugging
        .addTo(controller)
    );
  }, []);

  if (small) {
    return (
      <>
        <nav className="navbar navbar-expand-sm navbar-light fixed-top bg-white p-3">
          {/* Brand */}
          <Link to="/" className="navbar-brand">
            <img
              className="d-inline-block align-middle"
              src={logo}
              width="115"
              height="53"
              alt="NEXT by Pacific Life"
            />
          </Link>

          {/* Burger menu */}
          <HamButton
            type="button"
            toggler={toggle}
            clicked={clicked}
            aria-label="Toggle navigation"
          />

          {/* <HamburgerButton
            type="button"
            aria-label="Toggle navigation"
            toggler={toggle}
            clicked={clicked}
          /> */}
        </nav>

        {clicked ? (
          <div className="overlay text-right">
            <ExactNavLink to="/" onClick={toggle}>
              Home
            </ExactNavLink>
            <ExactNavLink to="/about" onClick={toggle}>
              About
            </ExactNavLink>
            <ExactNavLink to="/contact" onClick={toggle}>
              Contact
            </ExactNavLink>
          </div>
        ) : null}
      </>
    );
  }

  return (
    <nav className="navbar navbar-expand-sm navbar-light fixed-top p-3">
      {/* Brand */}
      <Link to="/" className="navbar-brand">
        <img
          className="d-inline-block align-middle"
          src={logo}
          width="115"
          height="53"
          alt="NEXT by Pacific Life"
        />
      </Link>

      {/* Links */}
      <div className="navbar-collapse collapse">
        <div className="mx-auto" />
        <ExactNavLink to="/" onClick={toggle}>
          Home
        </ExactNavLink>
        <ExactNavLink to="/about" onClick={toggle}>
          About
        </ExactNavLink>
        <ExactNavLink to="/contact" onClick={toggle}>
          Contact
        </ExactNavLink>
      </div>
    </nav>
  );
};

export default NavBar;
