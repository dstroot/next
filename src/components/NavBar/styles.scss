.navbar {
  // box-shadow: 0 2px 40px 0 rgba(0, 0, 0, 0.1);
  // text-transform: uppercase;
  // height: 110px;

  // use rgba values for clean animation
  border-bottom: 1px solid rgba(200, 200, 200, 0);
  background-color: rgba(255, 255, 255, 0);

  a {
    // text-transform: uppercase;
    font-weight: 400;
    color: gray('800');
    font-size: 24px;
    border-bottom: 2px solid rgba(0, 0, 0, 0);
    padding-bottom: 2px;
    &:hover {
      color: theme-color('primary');
    }
  }
  a.active {
    // color: theme-color('primary');
    border-bottom: 2px solid theme-color('primary');
  }

  .show a {
    // border-top: 1px solid rgba(0, 0, 0, 1);
    border-left: 1px solid rgba(200, 200, 200, 1);
    border-right: 1px solid rgba(200, 200, 200, 1);
    border-bottom: 1px solid rgba(200, 200, 200, 1);
  }
  .show a.active {
    color: theme-color('primary');
  }
}

.overlay {
  // display: block;
  // height: 100vh;
  // transition: all 0.4s linear;
  // background: gray;

  padding-top: 100px;

  a {
    font-weight: 400;
    background: white;
    color: gray('800');
    font-size: 28px;
    padding: 10px;
    padding-right: 20px;
    border-bottom: 1px solid rgba(200, 200, 200, 1);
    &:hover {
      color: theme-color('primary');
    }
  }
  a.active {
    color: theme-color('primary');
  }
}
