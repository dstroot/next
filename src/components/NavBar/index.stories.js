import React from 'react';
import { storiesOf } from '@storybook/react';

// components
import NavBar from '../NavBar';

// NavBar section
storiesOf('NEXT|NavBar', module)
  .addParameters({
    // default for all stories in this book
    info: {
      inline: true,
      header: false,
      text: `
        NavBar provides a navigation component that uses "@reach/router" for links. 
        It overrides some Bootstrap styles so it's "styles.scss" needs to be imported after Bootstrap in "/scss/main.scss".
        `,
    },
  })
  .add('show NavBar', () => <NavBar />);
