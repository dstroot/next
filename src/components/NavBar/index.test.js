import React from "react";
import renderer from "react-test-renderer";

import NavBar from "../NavBar";

describe("NavBar", () => {
  it("it should render", () => {
    const component = renderer.create(<NavBar />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
