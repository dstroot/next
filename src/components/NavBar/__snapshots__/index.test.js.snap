// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar it should render large 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-light fixed-top bg-white p-3"
>
  <a
    aria-current="page"
    className="navbar-brand"
    href="/"
    onClick={[Function]}
  >
    <img
      alt="NEXT by Pacific Life"
      className="d-inline-block align-middle"
      height="53"
      src="NXT_Logo_light.svg"
      width="115"
    />
  </a>
  <div
    className="navbar-collapse collapse"
  >
    <div
      className="mx-auto"
    />
    <a
      aria-current="page"
      className="active nav-link"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      className="nav-link"
      href="/about"
      onClick={[Function]}
    >
      About
    </a>
    <a
      className="nav-link"
      href="/contact"
      onClick={[Function]}
    >
      Contact
    </a>
  </div>
</nav>
`;

exports[`NavBar it should render small 1`] = `
<div
  width="400px"
>
  <nav
    className="navbar navbar-expand-sm navbar-light fixed-top bg-white p-3"
  >
    <a
      aria-current="page"
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="NEXT by Pacific Life"
        className="d-inline-block align-middle"
        height="53"
        src="NXT_Logo_light.svg"
        width="115"
      />
    </a>
    <div
      className="navbar-collapse collapse"
    >
      <div
        className="mx-auto"
      />
      <a
        aria-current="page"
        className="active nav-link"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        className="nav-link"
        href="/about"
        onClick={[Function]}
      >
        About
      </a>
      <a
        className="nav-link"
        href="/contact"
        onClick={[Function]}
      >
        Contact
      </a>
    </div>
  </nav>
</div>
`;
